src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -133,6 +133,11 @@ private boolean hasInvalidTimestamp(TransactionViewModel transactionViewModel) {
         // ignore invalid timestamps for transactions that were requested by our node while solidifying a milestone
         if(transactionRequester.wasTransactionRecentlyRequested(transactionViewModel.getHash())) {
             transactionRequester.removeRecentlyRequestedTransaction(transactionViewModel.getHash());
+            // as the transaction came from the request queue, we can add its branch and trunk to the request
+            // queue already, as we only have transactions in the request queue which are needed for solidifying
+            // milestones. this speeds up solidification significantly
+            transactionRequester.requestTrunkAndBranch(transactionViewModel);
```

Comment:
Kinda bothers me that `hasInvalidTimestamp` secretly requests transactions (one wouldn't expect that). I guess it is in the spirit of what is happening in `checkSolidity`... But it is still ugly

Maybe it makes sense to move it to the `ReceivedStage`?

Reply:
I'd need to check twice whether the transaction was in the recently requested queue if we want to do that.

Reply:
yeah I know...
It is an O(1) check
I agree that it is less efficient but it is a design that makes better sense.

I just don't think that this will be the bottleneck ever.

Another option is to move the logic up to `runValidation`.
It is still not clean that `runValidation` requests it, but it is better than to have it in `hasInvalidTimestamp`.

In short, I prefer to move it to the `ReceivedStage` (because I am pretty sure it is a small inefficiency) but can settle on `runValidation`.

Reply:
Alright, gonna move it.

Reply:
Moved to ReceivedStage.